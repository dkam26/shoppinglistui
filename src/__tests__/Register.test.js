import React from 'react';
import { shallow, mount } from 'enzyme';
import {configure} from 'enzyme';
import Register from '../components/register.component';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { shallowToJson } from 'enzyme-to-json';
configure({ adapter: new Adapter() });

describe('component', () => {
    const wrapper =  shallow(<Register />);
    it('should run', () => {
        expect(wrapper.exists(<div></div>)).toBe(true);
    });
    it('should render correctly', () => {
        const output = shallow( <Register />);
        expect(shallowToJson(output)).toMatchSnapshot();
        });
        it('render the two inputs', () =>{
            expect(wrapper.find("#surname")).toHaveLength(1);
            expect(wrapper.find("#firstname")).toHaveLength(1);
            expect(wrapper.find("#username")).toHaveLength(1);
            expect(wrapper.find("#password")).toHaveLength(1);
            expect(wrapper.find("#email")).toHaveLength(1);
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        });
})



