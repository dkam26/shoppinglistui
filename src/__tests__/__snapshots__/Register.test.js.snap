// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component render the two inputs 1`] = `
<div
  className="login-form"
>
  <style>
    
      body &gt; div,
      body &gt; div &gt; div,
      body &gt; div &gt; div &gt; div.login-form {
        height: 100%;
      }
    
  </style>
  <Grid
    style={
      Object {
        "height": "100%",
      }
    }
    textAlign="center"
    verticalAlign="middle"
  >
    <GridColumn
      style={
        Object {
          "maxWidth": 450,
        }
      }
    >
      <Header
        as="h2"
        color="teal"
        textAlign="center"
      >
        Register
      </Header>
      <Form
        as="form"
        size="large"
      >
        <Segment
          stacked={true}
        >
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            id="surname"
            name="surname"
            onChange={[Function]}
            placeholder="Surname"
          />
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            id="firstname"
            name="firstname"
            onChange={[Function]}
            placeholder="First name"
          />
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Username"
          />
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="lock"
            iconPosition="left"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
          />
          <Button
            as="button"
            color="blue"
            fluid={true}
            onClick={[Function]}
            size="large"
          >
            Create Account
          </Button>
          <div
            style={
              Object {
                "padding": "17px",
              }
            }
          >
            <Icon
              as="i"
              name="arrow circle outline left"
              onClick={[Function]}
              size="large"
            />
            Back
          </div>
        </Segment>
      </Form>
    </GridColumn>
  </Grid>
  <_class
    options={Object {}}
  />
</div>
`;

exports[`component should render correctly 1`] = `
<div
  className="login-form"
>
  <style>
    
      body &gt; div,
      body &gt; div &gt; div,
      body &gt; div &gt; div &gt; div.login-form {
        height: 100%;
      }
    
  </style>
  <Grid
    style={
      Object {
        "height": "100%",
      }
    }
    textAlign="center"
    verticalAlign="middle"
  >
    <GridColumn
      style={
        Object {
          "maxWidth": 450,
        }
      }
    >
      <Header
        as="h2"
        color="teal"
        textAlign="center"
      >
        Register
      </Header>
      <Form
        as="form"
        size="large"
      >
        <Segment
          stacked={true}
        >
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            id="surname"
            name="surname"
            onChange={[Function]}
            placeholder="Surname"
          />
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            id="firstname"
            name="firstname"
            onChange={[Function]}
            placeholder="First name"
          />
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Username"
          />
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="lock"
            iconPosition="left"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
          />
          <Button
            as="button"
            color="blue"
            fluid={true}
            onClick={[Function]}
            size="large"
          >
            Create Account
          </Button>
          <div
            style={
              Object {
                "padding": "17px",
              }
            }
          >
            <Icon
              as="i"
              name="arrow circle outline left"
              onClick={[Function]}
              size="large"
            />
            Back
          </div>
        </Segment>
      </Form>
    </GridColumn>
  </Grid>
  <_class
    options={Object {}}
  />
</div>
`;
